modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -1720,13 +1720,17 @@ protected Object keyValue(boolean cpy) {
 
             drReplicate(drType, null, newVer, topVer);
 
-            if (metrics && cctx.statisticsEnabled()) {
+            if (metrics && cctx.statisticsEnabled() && tx != null) {
```

Comment:
This code can move down where the checking of the transaction is already, but this way is also correct. Your coice.